### PR TITLE
2210 ASC order should not allow the see new posts button to appear

### PR DESCRIPTION
**Pull request by rowasc**
This pull request makes the following changes:
- do not get new posts when sorting by ASC Order

### Testing checklist:

Test normal behavior: 
- [x] Browser 1: load data view. Don't use sorting options
- [x] Browser 2: create and publish a new post
- [x] Browser 1 : wait ˜ 30 seconds and see the "see new posts"button appear.

Test new edge case: 
- [x] Browser 1: load data view. Use "oldest first" as your sort order
- [ ] Browser 2: create and publish a new post
- [x] Browser 1 : wait >30 seconds and don't see the "See new posts" button appear

Fixes ushahidi/platform#2210 .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rowasc** [comment]
Passed test locally (also it's easier to test if you use 3 browsers, but I didn't want to overcomplicate the checklist ha) 

@crcommons LMK

**Comment by crcommons** [comment]
When you are in normal data mode, add a new post and wait until the "See new posts" button appears. Then select sort by: oldest first. The button remains :(

**Comment by rowasc** [comment]
ah . Checking if I understand: 
- load data view, no sort order. 
- add post, see new posts buttons appear
- then without "removing " the button by clicking on it, change order? And that should hide the button ?

**Comment by crcommons** [comment]
Right, exactly.

**Comment by rowasc** [comment]
ugh :( ok , thanks !!!!

**Comment by rowasc** [comment]
@crcommons please review.

File: app/main/posts/views/post-view-data.directive.js [hunk 1]
```diff
@@ -350,8 +350,7 @@ function PostViewDataController(
     function getNewPosts() {
         var existingFilters = PostFilters.getQueryParams($scope.filters);
         var filterDate = moment(existingFilters.date_before).utc().format();
-
-        if (filterDate >= newPostsAfter) {
+        if (filterDate >= newPostsAfter && existingFilters.order !== 'asc') {
```
Comment (rjmackay) [inline app/main/posts/views/post-view-data.directive.js]: It might be a little clearer if this was split into its own check at the start of the fn.
```
if (existingFilters.order !== 'asc') {
    $scope.newPostsCount = 0;
}
```

**Comment by rjmackay** [comment]
Tested: 
- [x] Normal behaviour works
- [x] Sorting by asc works as expected
- [ ] Still need to test switching sort order after 'load new' is visible

I found another weird case:
- Load the data view
- Switch to sort: ASC
- Add a new post P1 (in other browser)
- Wait a while
- Switch to sort: desc. List loads, including the new post (P1)
- Add another post P2 (in other browser)
- Wait a while. 'See 2 new posts' appears
- Clicking on 'See 2 new posts' add 2 new posts adds P1 and P2 to the post list. **P1 is now in the list twice.**

**Comment by rjmackay** [comment]
Tested this too:

- load data view, no sort order.
- add post, see new posts buttons appear
- then without "removing " the button by clicking on it, change order

Works but the 'See 1 new post' button is hidden before I click 'Apply filters'.

**Comment by rjmackay** [comment]
@rowasc I think this works now. Could you or @Angamanga retest this and review my changes?

**Comment by Angamanga** [comment]
Tested locally and it now works:
- [ ] normal behaviour as described in first checklist
- [ ] switching to ask when button is visible makes the button disappear
- [ ] Only unloaded posts are visible after changing order from asc to desc and then posting another post are visible

**Comment by jshorland** [comment]
This passes QA in sandbox.uchaguzi.or.ke
